### PR TITLE
[ADD] web_widget_domain_editor_dialog: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Since v11 introduced the new domain editor widget it's not possible to edit
the selected records from the current domain.

This module reintroduces that dialog to complement the current widget with the
powerful search engine of Odoo.

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Some screenshots or GIFs would be very interesting to have in the README + generate it.

File: web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.js
Comment (yajo) [inline web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.js]: I'd call this module `web_wiget_domain_editor_dialog.DomainEditorDialog` and return that object directly.

Another option would be to have a single `web_widget_domain_editor_dialog` js module that performs all actions and returns the dict as it is doing now here.

**Review by yajo — APPROVED** [review]
I'm becoming used to attach gifs to Github and then link them directly from there. This way we don't need to keep those gifs in git, and they are visible anywhere also.

All things are suggestions, so OK anyways. Code & functional review.

**Comment by pedrobaeza** [comment]
@Yajo @chienandalu about the GIFs/images, the README generator includes code for unraveling its path and put an absolute one for not having problem with relative paths. Are you aware of that?

**Comment by yajo** [comment]
I'm not aware, but do you think it's better than using a free CDN such as github's?

**Comment by hbrunn** [comment]
using CDNs is a problem for deployments that can't access the internet

**Comment by pedrobaeza** [comment]
It uses GitHub as well, through raw.githubusercontent.com. See the initial commit for this:

https://github.com/OCA/maintainer-tools/pull/397/commits/77ed2b229ee9e63c9a0ef52b71a9705c48617636

**Comment by pedrobaeza** [comment]
@hbrunn too late...

**Comment by hbrunn** [comment]
ah, this is just about the readme. There I think it's fine, but all resources the module needs for functioning it should bring with it

**Comment by pedrobaeza** [comment]
@hbrunn yes, of course libs and so on are still bundled with the module itself. That hasn't changed and we also have that clear.

File: web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.js
Comment (chienandalu) [reply web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.js]: There you are :slightly_smiling_face:

**Comment by chienandalu** [comment]
All comments attended, thanks for the hints :smile:

**Comment by chienandalu** [comment]
@pedrobaeza I must be doing something wrong or do the image get uploaded to the CDN by the bot once the module is merged

**Comment by pedrobaeza** [comment]
@chienandalu the absolute path is intended for being used when the module is merged. Here in the PR, they are broken links.

**Comment by chienandalu** [comment]
Perfect then :smile:

**Review by yajo — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
